### PR TITLE
feat(plugin): specify plugin version

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
This PR makes it possible to install plugins with a specified version as below:

```bash
$ trivy plugin install referrer@v0.3.0
```

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by DmitriyLewen — COMMENTED** [review]
LGTM

Left 1 comment

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -88,17 +89,18 @@ func Update(ctx context.Context) error                  { return defaultManager(
...
 // Install installs a plugin
-func (m *Manager) Install(ctx context.Context, name string, opts Options) (Plugin, error) {
-	src := m.tryIndex(ctx, name)
+func (m *Manager) Install(ctx context.Context, arg string, opts Options) (Plugin, error) {
+	input := m.parseArg(arg)
+	input.name = m.tryIndex(ctx, input.name)
 
 	// If the plugin is already installed, it skips installing the plugin.
-	if p, installed := m.isInstalled(ctx, src); installed {
+	if p, installed := m.isInstalled(ctx, input.name); installed {
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/plugin/manager.go]: Added https://github.com/aquasecurity/trivy/pull/6683/commits/0b1556d0a9bba57b57d7b254c7b83d56de6e535b

**Comment by DmitriyLewen** [comment]
@knqyf263 i updated CLI docs after https://github.com/aquasecurity/trivy/pull/6683/commits/448bc4f85a5dfff8dead5cb728a5b8355c51de01

Can you also check macOS error? https://github.com/aquasecurity/trivy/actions/runs/9107705101/job/25037061764?pr=6683#step:10:8275

**Comment by knqyf263** [comment]
> @DmitriyLewen I was struggling with an error occurring only on GitHub Actions macOS runner, but finally addressed it. It's ready for review.

I made the same mistake 😔  Hopefully, it will pass tests this time 🤞

**Comment by knqyf263** [comment]
@DmitriyLewen Passed!

**Review by DmitriyLewen — APPROVED** [review]

**Pull request merged**